examples/CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
 project(libtorrent-examples)
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.11)
```

Comment:
I notice that this is newer than what's packaged on ubuntu, at least as of a few weeks ago. iirc, ubuntu has 3.10. Is the latest version important?

Reply:
It is even higher now (3.12). This one is available in Ubuntu 18.10, but why do I need to care about Ubuntu at all? There are backports available for versions down to 14.04 (e.g. the PPA for Travis builds in the master branch), so what's the problem?

Reply:
well, all I know is that I installed the most recent ubuntu about 3 months ago, and if I type `sudo apt install cmake`, I get an older version.
So, the trouble to pull down a build from cmake.org, is it warranted by some new feature?

Reply:
`list(TRANSFORM)` is why I raised the requirement. I felt uncomfortable because of a generator expression used earlier (which was nevertheless better than a ugly loop used before that). Generator expressions are evaluated after configuration stage (when generating build system files) and thus the final source list was not available during configuration stage. That looked incorrectly to me. 

The alternative way would be to put CMakeLists.txt files into the source subdirectories, call `target_sources` there and leave only `add_subdirectory()` call in the top-level `CMakeLists.txt` file. If you want, I can make these changes.

Reply:
It doesn't compile under latest Ubuntu 18.04.x LTS which comes with cmake 3.10.x.

Reply:
@zywo, I'm well aware of that. Do you believe it has to be buildable with the default set of Ubuntu 18.04 repositories? Otherwise you could use a PPA with a fresh cmake. For example, our Travis builds use ppa:dluxen/cmake-backports.

Reply:
I'm not going to go into detail but definitely should care about making this as compatible as possible.

It does not sound that the version change is absolutely necessary so please reconsider this as relying on a PPA is not desirable and likely to be an issue beyond Ubuntu.

Reply:
@zywo, @cas-- , I'll revert the changes that require cmake 3.11+.

Reply:
Done.